scanner/src/main/kotlin/ScanResultsStorage.kt

```diff
@@ -333,6 +363,30 @@ abstract class ScanResultsStorage {
         return Success(ScanResultContainer(pkg.id, scanResults))
     }
 
+    /**
+     * Internal version of [read] that does not update the [access statistics][stats]. The default implementation uses
+     * [Dispatchers.IO] to run requests for individual packages in parallel. Implementations may want to override this
+     * function if they can filter for the wanted [scannerCriteria] or fetch results for multiple packages in a more
+     * efficient way.
+     */
+    protected open fun readInternal(
+        packages: List<Package>,
+        scannerCriteria: ScannerCriteria
+    ): Result<Map<Identifier, List<ScanResult>>> {
+        val results = runBlocking(Dispatchers.IO) {
+            packages.map { async { readInternal(it, scannerCriteria) } }.map { it.await() }
```

Comment:
Good point, however, in the updated implementation I'm now using `flatMap { it.await() }` which I find a bit better than `awaitAll().flatten()`.

Reply:
The code you refer to is in `PostgresStorage`; here it would be possible to use `awaitAll()`; but it is fine with me, it does not make much of a difference.

Reply:
Right, I missed that. Done.